apps/app-backend/src/utils/azure-search/health-check.ts

```diff
@@ -0,0 +1,20 @@
+import { pipe } from "fp-ts/lib/function";
+
+import * as TE from "fp-ts/lib/TaskEither";
+
+import {
+  HealthCheck,
+  toHealthProblems,
+} from "@pagopa/io-functions-commons/dist/src/utils/healthcheck";
+import { AzureSearchClientDependency } from "./dependency";
+
+export type AzureSearchProblemSource = "AzureSearch";
+
+export const makeAzureSearchHealthCheck = <T>({
+  searchClient,
+}: AzureSearchClientDependency<T>): HealthCheck<AzureSearchProblemSource> =>
+  pipe(
+    searchClient.fullTextSearch({ searchText: "test" }),
+    TE.mapLeft(toHealthProblems("AzureSearch" as const)),
```

Comment:
**nit**: not needed for this PR but generally it's better to call conversion methods with the prefix "from" (instead that "to"). why? because in this way you can have multiple `fromOriginalType` methods (ie. `fromX(x: X)` `fromY(y: Y)`) **in the same module**: the module of the target type; while you can have _only one_ `toX` method in the module.